src/index.js

```diff
@@ -15,12 +17,33 @@ const app = express();
 const server = new ApolloServer({
   ...schema,
   resolvers,
-  context: { models },
+  context: async ({ req, res }) => ({ models, user: req.user }),
   instrospection: true,
   playground: true,
   tracing: true,
 });
 
+app.use(function (req, res, next) {
+  passport.authenticate('jwt', { session: false }, (err, user, info) => {
+    req.user = user;
+    next();
+  })(req, res, next);
+});
+
+var JwtStrategy = require('passport-jwt').Strategy,
+  ExtractJwt = require('passport-jwt').ExtractJwt;
+var opts = {};
+opts.jwtFromRequest = ExtractJwt.fromAuthHeaderAsBearerToken();
+opts.secretOrKey = process.env.SECRET;
+passport.use(
+  new JwtStrategy(opts, async function (jwt_payload, done) {
+    return done(
+      null,
+      await models.user.findOne({ where: { id: jwt_payload.sub } }),
+    );
+  }),
+);
+
```

Comment:
You can extract all this Passport initialization logic into its own module, import it and use it like:

```suggestion
app.use(passportMiddleware)
```

Since `app.use()` receives a function as a parameter, you can create a new module that exports a function by default , like:

```
export default (req, res, next) => {
  passport.authenticate('jwt', { session: false }, (err, user, info) => {
    req.user = user;
    next();
  })(req, res, next);
}
```

Reply:
Solved. Clean Code +1000000000 :+1: :100: